core/download.ts

```diff
@@ -1,9 +1,11 @@
+import type { JsonFeed } from './loader/json-feed.js'
 import { request } from './request.js'
 
 export interface TextResponse {
   readonly headers: Headers
   readonly ok: boolean
   parse(): Document | XMLDocument
+  parseJson(): JsonFeed | null
```

Comment:
`TextResponse` is a general type to load any data.

We should not use exact JSON schema here. Let’s just return the common `object` and type in loader after we checked the format.